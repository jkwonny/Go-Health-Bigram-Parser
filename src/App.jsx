import React, {Component} from 'react';
import Particles from 'react-particles-js';
import style from './index.css';
import path from 'path';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      histogram: {},
      uploaded: false
    }
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  //post request sending filename and receiving histogram data
  //change state to conditionally render histogram
  async handleSubmit () {
    event.preventDefault();
    let browser_file_path = document.getElementById("myfileid").value;
    browser_file_path = browser_file_path.slice(12,browser_file_path.length)
    browser_file_path = path.join(__dirname + browser_file_path);

    const settings = {
      method: 'POST',
      headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
      },
      body: JSON.stringify({ browser_file_path: browser_file_path})
  };
    try {
      const fetchResponse = await fetch('/bigram', settings);
      const data = await fetchResponse.json();
      this.setState({ histogram: data.histogram })
      this.setState({ uploaded: true })
      return data;
    } catch(e) {
      return e;
    }
}


  render (){
    let finalHistogram = [];
    if (this.state.uploaded === true) {
      for (let [key, value] of Object.entries(this.state.histogram)) {
        finalHistogram.push(<ul>{`"${key}"`} {value}</ul>)
      }
    }

  return (
    <div className="fullscreen">
        <div className="decor">
          <Particles
            params={{
              particles: {
                line_linked: {
                  enable: true,
                  distance: 150,
                  color: "#74e1da",
                  opacity: 0.7,
                  width: 1
                },
                number: {
                  value: 100
                },
                size: {
                  value: 1
                }
              },
              interactivity: {
                events: {
                  onhover: {
                    enable: true,
                    mode: 'repulse'
                  }
                }
              }
            }}
          />
        </div>
        <header className="bigramParser">
          <h1 className="welcomeText">Welcome GoHealth</h1>
          <div className="uploadContainer">
            <p className="instructionText">Please import the desired file you'd like to test</p>
            <div className="upload-btn-wrapper">
              <form className="formcontainer" onSubmit={this.handleSubmit}>
                <input type="file" id="myfileid" name="myfile" accept=".txt, .doc, .docx, .pdf"/>
                <input className="submitButton" type="submit" value="Submit"></input>
              </form>
            </div>
          </div>
        </header>
        <div className="histogram">
          {finalHistogram}
        </div>
    </div>
  );
}
}

export default App;
