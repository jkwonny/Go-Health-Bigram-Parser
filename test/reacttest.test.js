import App from "../src/App.jsx";
import React, {Component} from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import "babel-polyfill"

//enzyme configuration
configure({ adapter: new Adapter() });

//testing App component
describe("testing App component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<App />); 
  });
  it("renders without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });
  it("matches snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should render a two input types", () => {
    expect(wrapper.find('input')).toHaveLength(2);
  })
});