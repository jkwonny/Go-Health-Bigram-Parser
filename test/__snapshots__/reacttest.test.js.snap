// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App component matches snapshot 1`] = `
<div
  className="fullscreen"
>
  <div
    className="decor"
  >
    <t
      height="100%"
      params={
        Object {
          "interactivity": Object {
            "events": Object {
              "onhover": Object {
                "enable": true,
                "mode": "repulse",
              },
            },
          },
          "particles": Object {
            "line_linked": Object {
              "color": "#74e1da",
              "distance": 150,
              "enable": true,
              "opacity": 0.7,
              "width": 1,
            },
            "number": Object {
              "value": 100,
            },
            "size": Object {
              "value": 1,
            },
          },
        }
      }
      style={Object {}}
      width="100%"
    />
  </div>
  <header
    className="bigramParser"
  >
    <h1
      className="welcomeText"
    >
      Welcome GoHealth
    </h1>
    <div
      className="uploadContainer"
    >
      <p
        className="instructionText"
      >
        Please import the desired file you'd like to test
      </p>
      <div
        className="upload-btn-wrapper"
      >
        <form
          className="formcontainer"
          onSubmit={[Function]}
        >
          <input
            accept=".txt, .doc, .docx, .pdf"
            id="myfileid"
            name="myfile"
            type="file"
          />
          <input
            className="submitButton"
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </header>
  <div
    className="histogram"
  />
</div>
`;
